pkg/core/transaction/transaction.go

```diff
@@ -107,6 +110,7 @@ func (t *Transaction) AddVerificationHash(addr util.Uint160) {
 func (t *Transaction) DecodeBinary(br *io.BinReader) {
 	t.Type = TXType(br.ReadB())
 	t.Version = uint8(br.ReadB())
+	t.Nonce = uint32(br.ReadB())
```

Comment:
`Nonce` is 4-bytes, while `ReadB()` reads one.
You should use `br.ReadU32LE` (or `BE`) as in reference implementation.